model/src/lectures/lectureDoc.ts

```diff
@@ -31,16 +31,27 @@ namespace LectureDisplay {
         quizModel: Quizzes.Quiz;
     }
 
+    export interface Question {
+        text: string,
+        //userID: string,
+        anonym: boolean,
+        likes: number,
+        postedTime: Timestamp,
+    }
+
     export interface Lecture {
         pdfUri: string;
         nbOfPages: number;
         availableToStudents: boolean;
         audioTranscript: { [pageNumber: number]: string; };
+        questions: Question[];
     }
 
     export const Functions = FunctionFolder("lectures", {
         addAudioTranscript: FunctionOf<{ courseId: string, lectureId: string, pageNumber: number, transcription: string; }, {}, 'invalid_arg' | 'error_firebase' | 'successfully_added'>("addAudioTranscript"),
+        createQuestion: FunctionOf<{ courseId: string, lectureId: string, question: Question; }, { id: string; }, 'invalid_arg' | 'error_firebase' | 'empty_question'>("createQuestion"),
```

Comment:
The cloud function argument `question` being passed on "as it is" is a potential breach of security as it is directly written to the database.
For now since the questions are only texts, maybe take only a string argument, sanitize it and construct a Question object in the cloud function might be a more secure way of handling it.

Reply:
Yes thank you, you are absolutely right